### PR TITLE
fix: Sys Diagram clipped if wider than container

### PR DESCRIPTION
**Pull request by deepak-bhardwaj-ps**
Signed-off-by: Deepak Bhardwaj <Deepak.Bhardwaj@outlook.com>

## Hey, I just made a Pull Request!

Fixed the issue where system diagram was getting clipped when its width was larger than its container.

Screenshot 2021-06-25 at 9 14 01 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Nice! Gonna need a changeset on this too (`yarn changeset`).

Does it still look ok for small graphs too or does it get huge? No biggie, just checking

**Comment by deepak-bhardwaj-ps** [comment]
> Nice! Gonna need a changeset on this too (`yarn changeset`).
> 
> Does it still look ok for small graphs too or does it get huge? No biggie, just checking

Verified small graphs as well, looks pretty well. The difference is, it will be centre aligned on the screen.

File: packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ export function DependencyGraph({
     <svg
       ref={containerRef}
       {...svgProps}
-      width={maxWidth}
+      width="100%"
       height={maxHeight}
```
Comment (timbonicus) [inline packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx:266]: Probably makes sense to set height to `100%` too?

File: plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx [hunk 2]
```diff
@@ -262,6 +262,8 @@ export function SystemDiagramCard() {
         nodeMargin={10}
         direction={DependencyGraphTypes.Direction.BOTTOM_TOP}
         renderNode={RenderNode}
+        paddingX={50}
```
Comment (timbonicus) [inline plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx:265]: Seems slightly preferable to use theme spacing here, like:
```
const theme = useTheme();
...
return (
  <InfoCard title="System Diagram">
    <DependencyGraph
      paddingX={theme.spacing(4)}
      paddingY={theme.spacing(4)}
 ```

File: .changeset/grumpy-sloths-check.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core-components': patch
+'@backstage/plugin-catalog': patch
+---
+
+System Diagram was getting clipped when it was wider than the container.
```
Comment (timbonicus) [inline .changeset/grumpy-sloths-check.md:6]:
```suggestion
Fixed sizing of the System diagram when the rendered graph was wider than the container.
```
Sorry for the small nit here, just wanted to make the language sound more like a change than a problem 😅

File: packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx (hunk 1 above)
Comment (deepak-bhardwaj-ps) [reply packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx:266]: I tested with height 100% as well, that doesn't look good.

File: plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx (hunk 2 above)
Comment (deepak-bhardwaj-ps) [reply plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx]: Verified that it works well with theme spacing, updated in code.

**Comment by freben** [comment]
Seems you accidentally got some unrelated adopters and e2e stuff in your PR now!

**Comment by deepak-bhardwaj-ps** [comment]
Accepting change from Github UI somehow screwed up as it doesn't include signoff. Don't know how other changes came in from other committer, Fixed now.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**